tikv-client/src/main/java/com/pingcap/tikv/KVClient.java

```diff
@@ -0,0 +1,227 @@
+    this.conf = conf;
+    this.clientBuilder = clientBuilder;
+    //    ExecutorService executors = Executors.newFixedThreadPool(conf.getKVClientConcurrency());
+    ExecutorService executors = Executors.newFixedThreadPool(20);
+    this.completionService = new ExecutorCompletionService<>(executors);
+  }
+
+  @Override
+  public void close() {}
+
+  /**
+   * Get a raw key-value pair from TiKV if key exists
+   *
+   * @param key raw key
+   * @return a ByteString value if key exists, ByteString.EMPTY if key does not exist
+   */
+  public ByteString get(ByteString key, long version) {
+    BackOffer backOffer = ConcreteBackOffer.newGetBackOff();
+    while (true) {
+      RegionStoreClient client = clientBuilder.build(key);
+      try {
+        return client.get(backOffer, key, version);
+      } catch (final TiKVException e) {
+        backOffer.doBackOff(BackOffFunction.BackOffFuncType.BoRegionMiss, e);
+      }
+    }
+  }
+
+  /**
+   * Get a set of key-value pair by keys from TiKV
+   *
+   * @param keys keys
+   */
+  public List<KvPair> batchGet(List<ByteString> keys, long version) {
+    return batchGet(ConcreteBackOffer.newBatchGetMaxBackOff(), keys, version);
+  }
+
+  private List<KvPair> batchGet(BackOffer backOffer, List<ByteString> keys, long version) {
+    Set<ByteString> set = new HashSet<>(keys);
+    return batchGet(backOffer, set, version);
+  }
+
+  private List<KvPair> batchGet(BackOffer backOffer, Set<ByteString> keys, long version) {
+    Map<TiRegion, List<ByteString>> groupKeys = groupKeysByRegion(keys);
+    List<Batch> batches = new ArrayList<>();
+
+    for (Map.Entry<TiRegion, List<ByteString>> entry : groupKeys.entrySet()) {
+      appendBatches(batches, entry.getKey(), entry.getValue(), BATCH_GET_SIZE);
+    }
+    return sendBatchGet(backOffer, batches, version);
+  }
+
+  /**
+   * Scan raw key-value pairs from TiKV in range [startKey, endKey)
+   *
+   * @param startKey raw start key, inclusive
+   * @param endKey raw end key, exclusive
+   * @return list of key-value pairs in range
+   */
+  public List<Kvrpcpb.KvPair> scan(ByteString startKey, ByteString endKey, long version) {
+    Iterator<Kvrpcpb.KvPair> iterator =
+        scanIterator(conf, clientBuilder, startKey, endKey, version);
+    List<Kvrpcpb.KvPair> result = new ArrayList<>();
+    iterator.forEachRemaining(result::add);
+    return result;
+  }
+
+  /**
+   * Scan raw key-value pairs from TiKV in range [startKey, endKey)
+   *
+   * @param startKey raw start key, inclusive
+   * @param limit limit of key-value pairs
+   * @return list of key-value pairs in range
+   */
+  public List<Kvrpcpb.KvPair> scan(ByteString startKey, int limit) {
+    Iterator<Kvrpcpb.KvPair> iterator = scanIterator(conf, clientBuilder, startKey, limit);
+    List<Kvrpcpb.KvPair> result = new ArrayList<>();
+    iterator.forEachRemaining(result::add);
+    return result;
+  }
+
+  /** A Batch containing the region, a list of keys to send */
```

Comment:
s/A Batch containing the region, a list of keys to send/A Batch containing the region and a list of keys to send to the region

Reply:
ok